File: 240711/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.py
m = int(input())
a, b = map(int, input().split())

def binary_sort(target):
    
    start = 1
    end = m
    cnt = 0
    while start <= end:
        
        mid = (start + end) // 2
        cnt += 1
        if mid == target:
            return cnt
        elif mid > target:
            end = mid - 1
        else:
            start = mid + 1

    return cnt


min_ans = float('inf')
max_ans = 0
for i in range(a, b+1):
    x = binary_sort(i)

    min_ans = min(min_ans, x)
    max_ans = max(max_ans, x)

print(min_ans, max_ans)

File: 240711/컴퓨터와 함께하는 숫자 게임 2/play-number-game-with-computer-2.js
const fs = require('fs')
const input = fs.readFileSync(0).toString().trim().split("\n")

const m = Number(input[0])
const [a, b] = input[1].split(" ").map(x => Number(x))

function binarySort(x) {

    let start = 1
    let end = m
    let cnt = 0

    while (start <= end) {
        let mid = parseInt((start + end) / 2)
        cnt += 1
        if (mid === x) {
            return cnt
        }
        else if (mid > x) {
           end = mid - 1 
        } else {
        start = mid + 1
        }
    }

    return cnt
}

let max = 0
let min = Infinity


for (let i = a; i <= b; i++) {
    res = binarySort(i)

    max = Math.max(max, res)
    min = Math.min(min, res)
}

console.log(min, max)